internal/cephfs/volumeoptions.go

```diff
@@ -299,10 +302,18 @@ func newVolumeOptionsFromVolID(ctx context.Context, volID string, volOpt, secret
 	}
 
 	volOptions.ProvisionVolume = true
-	volOptions.RootPath, err = getVolumeRootPathCeph(ctx, &volOptions, cr, volumeID(vid.FsSubvolName))
+	info, err = getSubVolumeInfo(ctx, &volOptions, cr, volumeID(vid.FsSubvolName))
 	if err != nil {
-		return &volOptions, &vid, err
+		if errors.Is(err, ErrInvalidCommand) {
+			volOptions.RootPath, err = getVolumeRootPathCeph(ctx, &volOptions, cr, volumeID(vid.FsSubvolName))
+			if err != nil {
+				return &volOptions, &vid, err
+			}
+			return &volOptions, &vid, nil
+		}
+		return &volOptions, &vid, status.Error(codes.Internal, err.Error())
 	}
+	volOptions.RootPath = info.Path
 	return &volOptions, &vid, nil
```

Comment:
Avoid multiple reaturns

```suggestion
info, err = getSubVolumeInfo(ctx, &volOptions, cr, volumeID(vid.FsSubvolName))
	// This will be overridden if the err is ErrInvalidCommand
	volOptions.RootPath = info.Path
	if err != nil {
		if errors.Is(err, ErrInvalidCommand) {
			volOptions.RootPath, err = getVolumeRootPathCeph(ctx, &volOptions, cr, volumeID(vid.FsSubvolName))
		}
	}

	if err != nil {
		return &volOptions, &vid, err
	}
	return &volOptions, &vid, nil
```

Reply:
Thanks, Will make the change.